### DISCUSSION TITLE
Best way to trigger validation when using useFieldArray functions?

### DISCUSSION THREAD

**Discussion by GollyJer** [react-hook-form/react-hook-form#9147] [Q&A] [answered]
I have a form...

```javascript 
  const form = useForm<FormValues>({
    mode: 'onChange',
    reValidateMode: 'onChange',
    defaultValues,
    resolver: yupResolver(validationRules),
    delayError: 600,
  });
```

When input values are changed, validation is run immediately. However, when using the `append` and `remove` functions of `useFieldArray`, I've resorted to calling `form.trigger()`. 

```javascript 
<Button
  onPress={() => {
    fieldArray.append({ amount: '', name: '' });
    form.trigger();
  }}
>
  +
</Button>
```

Is this the correct approach? Thanks!

**Comment by leapful** [comment]
@GollyJer : Could you please explain  why you need to call "form.trigger()" when appending or removing ?

**Reply by GollyJer** [reply]
To trigger form validation.

**Reply by GollyJer** [reply]
Maybe I worded this badly? The question I am asking...

How do you get a form to validate when calling fieldArray`.append`?

**Reply by leapful** [reply]
@GollyJer : As I remember, append or any action from useFieldArray would trigger validation on the affected field. Let me check it first.

**Reply by Moshyfawn** [reply]
For field validation in a field array, you need to `trigger` it on `append`, correct: [an example csb](https://codesandbox.io/s/bold-ishizaka-5kn6by?file=%2Fsrc%2FApp.tsx).

There was a proposal to add should["MethodName"] option to field array methods, but we haven't looked into it yet

**Reply by GollyJer** [reply]
@Moshyfawn. That confirms I'm doing it right. Thanks!

Can you post this as an answer and I'll mark it as accepted. 👍

**Comment by GollyJer** [comment, marked as answer]
From @Moshyfawn...

For field validation in a field array, you need to trigger it on append, correct: [an example csb](https://codesandbox.io/s/bold-ishizaka-5kn6by?file=%2Fsrc%2FApp.tsx).

There was a proposal to add should["MethodName"] option to field array methods, but we haven't looked into it yet.
